ddl/column.go

```diff
@@ -700,6 +706,8 @@ func needChangeColumnData(oldCol, newCol *model.ColumnInfo) bool {
 			return oldCol.Flen != newCol.Flen || oldCol.Decimal != newCol.Decimal || toUnsigned != originUnsigned
 		case mysql.TypeEnum, mysql.TypeSet:
 			return isElemsChangedToModifyColumn(oldCol.Elems, newCol.Elems)
+		case mysql.TypeTiny, mysql.TypeShort, mysql.TypeInt24, mysql.TypeLong, mysql.TypeLonglong:
+			return needTruncationOrToggleSignForInteger()
```

Comment:
Checking `toUnsigned != originUnsigned` is enough?

Reply:
I don't think so, if `newColFlen < oldColFlen`, we need reorg the data.

Reply:
But, can newColFlen not the same as oldColFlen?

Reply:
yes, for diff type in integer, change `bigint` to `int` is the case the decrease the flen.